src/averageAges.js

```diff
@@ -55,7 +77,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = people.filter(person => (
+    people.some(mother => mother.name === person.mother) && (
+      (onlyWithSon) ? person.sex === 'm' : true
+    )
+  ));
+
+  const ages = children.map(child => (
+    child.born - people.find(mother => mother.name === child.mother).born
+  ));
+
+  const sumAllAges = ages.reduce((prevValue, currentValue) => (
+    prevValue + currentValue
+  ));
+
+  return Math.round(sumAllAges / ages.length * 100) / 100;
```

Comment:
using it three times in your code - introduce a helper function that you can use across the code